src/main/java/org/thoughtslive/jenkins/plugins/jira/webhook/JiraIssueChangeListener.java

```diff
@@ -0,0 +1,21 @@
+package org.thoughtslive.jenkins.plugins.jira.webhook;
+
+public interface JiraIssueChangeListener {

```

Comment:
Notifications content are very different, especially if we consider sprint events, and transition events.
Also there are 2 levels of webhook handling:
 * the JSON event
 * the step triggered by the event.
For example, you can have 2 different steps triggered for the same event: the parsing should be the same, thus before calling the listener.

getEventKey() is a non sense for the create issue event: no one can predict the ID of a new issue.

In an agile point of view, wait for a user need to generalize :)

Reply:
I don't see any problem generalizing and moving the custom logic to actual steps, instead of maintaining everything in `JiraWebhook`, regardless of the type, key and JSON payload, every step would return key and the logic to filter the JSON message is also inside a step. 

I vote for generic things, like this case getEventKey() is not a non sense, key as project name, and actual filtering will be just checking if the create event key is prefixed with the same, as I said above the logic to filter is also going to stay with the step.